File: src/dataset/javascript/kindergarten-garden_exercism-ThomasZumsteg/claude/ClaudeSonnet4_kindergarten-garden_v2.js
var DEFAULT_STUDENTS = [ 'Alice', 'Bob', 'Charlie', 'David', 'Eve', 
		'Fred', 'Ginny', 'Harriet', 'Ileana', 'Joseph', 'Kincaid', 'Larry'];

var PLANT_MAP = {'C': 'clover', 'G': 'grass', 'R': 'radishes', 'V': 'violets'};

var Garden = function(garden, students) {
	/* A Kindergarden plant garden */
	students = students ? students.slice().sort() : DEFAULT_STUDENTS;
	var rows = garden.split('\n');
	var parsedRows = [
		MakeRow(rows[0]),
		MakeRow(rows[1])
	];

	for(var i = 0; i < students.length; i++) {
		this[students[i].toLowerCase()] = StudentsGarden(parsedRows, i);
	}
};

function MakeRow(row) {
	/* Converts a row of plant letters to an array of plants */
	var result = new Array(row.length);
	for(var i = 0; i < row.length; i++) {
		result[i] = PLANT_MAP[row[i]];
	}
	return result;
}

function StudentsGarden(garden, position) {
	/* The plants owned by an individual student */
	var startPos = 2 * position;
	return [
		garden[0][startPos],
		garden[0][startPos + 1],
		garden[1][startPos],
		garden[1][startPos + 1]
	];
}

module.exports = Garden;

File: src/dataset/python/kindergarten-garden_exercism-fruit-in/claude/ClaudeSonnet4_kindergarten_garden_v2.py
class Garden:
    STUDENTS = ["Alice", "Bob", "Charlie", "David", "Eve", "Fred",
                "Ginny", "Harriet", "Ileana", "Joseph", "Kincaid", "Larry", ]
    PLANTS = {"G": "Grass", "C": "Clover", "R": "Radishes", "V": "Violets", }

    def __init__(self, diagram, students=STUDENTS):
        row0, row1 = diagram.split('\n')
        students = sorted(students)
        self.cups = {}

        for i, student in enumerate(students):
            if i * 2 + 1 >= len(row0):
                break
            idx = i * 2
            seeds = [row0[idx], row0[idx + 1], row1[idx], row1[idx + 1]]
            self.cups[student] = [self.PLANTS[seed] for seed in seeds]

    def plants(self, student):
        return self.cups[student]